update desc of namespace defaulting in CLI (#28695)
kubernetes/website: content/en/docs/reference/kubectl/overview.md

If you need help, run `kubectl help` from the terminal window.

> ## Namespace Defaulting

Reviewer: Important to call out that this only applies to in-cluster auth - so a running pod using kubectl.

> ## In-Cluster Auth, Namespace Defaulting

Reviewer 2: ```suggestion
## In-cluster authentication and namespace overrides
```

or move the namespace override later as a 3rd level heading

> If you are running `kubectl` within a pod, and thus in a cluster, the cli attempts to scope CLI commands to the pod's namespace. 

Reviewer: Something like this.

```suggestion
By default `kubectl` will first determine if it is running within a pod, and thus in a cluster. It starts by checking for the `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` environment variables and the existence of a service account token file at `/var/run/secrets/kubernetes.io/serviceaccount/token`. If all three are found in-cluster authentication is assumed.

To maintain backwards compatibility, if the `POD_NAMESPACE` environment variable is set during in-cluster authentication it will override the default namespace from the from the service account token. Any manifests or tools relying on namespace defaulting will be affected by this.
```

> **POD_NAMESPACE Environment Variable**

Reviewer 2: ```suggestion
**`POD_NAMESPACE` environment variable**
```
Is this meant as a heading? If so, let's make that clear via Markdown.

If the `POD_NAMESPACE` environment variable is set, cli operations on namespaced resources will default to the variable value. For example, if the variable is set to `seattle`, `kubectl get pods` would return pods in the `seattle` namespace. This is because pods are a namespaced resource, and no namespace was provided in the command. Review the output of `kubectl api-resources` to determine if a resource is namespaced. 

Explicit use of `--namespace <value>` overrides this behavior. 

> **Service Account**

Reviewer 2: ```suggestion
**How kubectl handles ServiceAccount tokens**
```
(or as a heading)
